### PR TITLE
Switch language based on user agent

### PR DESCRIPTION
**Pull request by clementbiron**
That changeset:
- render default english website in the subdirectory `/en` and define all the necessary redirections, for exemple `https://opentermsarchive.org/privacy-policy` to `https://opentermsarchive.org/en/privacy-policy`
- redirects requested pages without any query language parameter depending on the browser's language preferences, for example `https://opentermsarchive.org` redirects to `https://opentermsarchive.org/fr/` with my browser configured in french, `https://opentermsarchive.org/impact` redirects to `https://opentermsarchive.org/en/impact` with my browser configured in english
 - adds tests to verify the added behaviours and runs them in CI

### CODE REVIEW

File: themes/opentermsarchive/layouts/alias.html [hunk 1]
```diff
@@ -1,11 +1,32 @@
 <!doctype html>
 <html lang="{{ site.Language.LanguageCode }}">
   <head>
-    <title>{{ .Permalink }}</title>
+    <title>{{ substr .Permalink 0 120 }}</title>
```
Comment (Ndpnt) [inline themes/opentermsarchive/layouts/alias.html:4]: Add comment to explain why

File: themes/opentermsarchive/layouts/alias.html [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+        {{ range .Translations }}
+          page.translations.push({ "permalink": "{{ .Permalink }}", "lang": "{{ .Language.LanguageCode }}" });
+        {{ end }}
+        pages.push(page);
+      {{ end }}
+    </script>
   </head>
-  <body></body>
+  <body>
+    <p>You should be redirected very quickly, if not <a href="{{ .Permalink }}">click here</a>.</p>
```
Comment (Ndpnt) [inline themes/opentermsarchive/layouts/alias.html:28]: i18n

File: test/cypress/e2e/i18n.cy.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      context('when window navigator languages is defined as [ja-JP]',() => {
+        beforeEach(function () {
+          cy.visit('/', {
+            onBeforeLoad (win) {
+              Object.defineProperty(win.navigator, 'languages', { value: ['ja-JP'] });
+            }
+          })
+        })
+        it(`is redirected to ${EN_HOMEPAGE}`, () => {
+          cy.task('log', 'This will be output to the terminal')
```
Comment (Ndpnt) [inline test/cypress/e2e/i18n.cy.js:34]: Remove log

File: test/cypress/e2e/i18n.cy.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          cy.location('href').then(href => {
+            cy.task('log', href)
```
Comment (Ndpnt) [inline test/cypress/e2e/i18n.cy.js:36]: Remove log

File: themes/opentermsarchive/layouts/alias.html (hunk 1 above)
Comment (clementbiron) [reply themes/opentermsarchive/layouts/alias.html:4]: As we saw together, I'm putting a translatable text in its place.

File: test/cypress/e2e/i18n.cy.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+          cy.location('href').should('equal', EN_HOMEPAGE)
+        });
+        it('<html> lang attribute value is equal to en', () => {
+          cy.get('html').should('have.attr', 'lang', 'en');
+        });
+        it('content is in english', () => {
+          cy.get('.footer').contains('Privacy policy');
+        });
+      });
+    });
```
Comment (Ndpnt) [inline test/cypress/e2e/i18n.cy.js]:
```suggestion
describe('Homepage', () => {
    context('when language is not specified in the URL', () => {
      context('when user prefered language is supported', () => {
        beforeEach(() => {
          cy.visit('/', {
            onBeforeLoad(win) {
              Object.defineProperty(win.navigator, 'languages', { value: ['fr-FR'] });
            },
          });
        });
        it('adds language in the URL', () => {
          cy.location('href').should('equal', FR_HOMEPAGE);
        });
        it('defines the proper HTML lang attribute', () => {
          cy.get('html').should('have.attr', 'lang', 'fr');
        });
        it('displays the content in the proper language', () => {
          cy.get('.footer').contains('Politique de confidentialité');
        });
      });
      context('when user prefered language is not supported', () => {
        beforeEach(() => {
          cy.visit('/', {
            onBeforeLoad(win) {
              Object.defineProperty(win.navigator, 'languages', { value: ['ja-JP'] });
            },
          });
        });
        it('adds the default language in the URL', () => {
          cy.location('href').should('equal', EN_HOMEPAGE);
        });
        it('defines the HTML lang attribute to the default language', () => {
          cy.get('html').should('have.attr', 'lang', 'en');
        });
        it('displays the content in the default language', () => {
          cy.get('.footer').contains('Privacy policy');
        });
      });
    });
```

File: test/i18n.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+waitOn({ resources: ['http://localhost:1313'] }, async (err) => {
+    if (err) {
+      console.error("Error when waiting for Hugo", err);
+      return hugoServer.kill('SIGTERM');
+    }
+
+    console.log('→ Start Cypress');
+    try {
+      const results = await cypress.run({
```
Comment (Ndpnt) [inline test/i18n.js:25]: `results` is not used here

File: test/i18n.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        configFile: 'test/cypress/cypress.config.js'
+      });
+    } catch (err) {
+      console.error('Cypress error', err);
+    } finally {
+      try {
+        hugoServer.kill('SIGTERM');
+      } catch (err) {
```
Comment (Ndpnt) [inline test/i18n.js:33]:
```suggestion
    } catch (error) {
      console.error('Error occurred while trying to kill the Hugo process:', error);
    }
```

File: test/i18n.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        console.error('Error when stopping Hugo', err);
+      }
```
Comment (Ndpnt) [inline test/i18n.js:35]:
```suggestion
      } catch (error) {
        console.error('Error occurred while trying to kill the Hugo process:', error);
      }
```

File: test/cypress/cypress.config.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+const { defineConfig } = require('cypress');
```
Comment (Ndpnt) [inline test/cypress/cypress.config.js:1]: add comment to explain that this code come from cypress doc

**Review by Ndpnt — APPROVED** [review]

**Pull request merged**